indexer/test/helpers/http-server.js

```diff
@@ -0,0 +1,25 @@
+import { once } from 'node:events'
+import http from 'node:http'
+
+/**
+ * @param {http.RequestListener} handler
+ */
+export async function givenHttpServer (handler) {
+  const server = http.createServer((req, res) => {
+    ;(async () => {
+      await handler(req, res)
+    })().catch(err => {
```

Comment:
```suggestion
    handler(req, res).catch(err => {
```

Reply:
I accepted your suggestion, but I need to tweak it as TypeScript complains:

```
Property 'catch' does not exist on type 'void'.
```

Solution:

```js
Promise.resolve(handler(req, res)).catch(err => {
```

There is one more subtle difference:
- In my code, if the `handler` function is sync and throws, the error is still handled.
- In your version, such situation triggers "uncaught error"

In that light, I am going to keep my original version.